### PR TITLE
Adding logging to make regional scenarios more clear

### PR DESCRIPTION
**Pull request by trwalke**
Was able to confirm the regional configuration api works. Logs make the used endpoint misleading. 
Fix for https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/6510

### CODE REVIEW

File: lib/msal-node/src/client/ClientApplication.ts [hunk 1]
```diff
@@ -412,10 +412,23 @@ export abstract class ClientApplication {
             : this.config.auth.azureCloudOptions;
 
         // using null assertion operator as we ensure that all config values have default values in buildConfiguration()
-        this.logger.verbose(
-            `building oauth client configuration with the authority: ${authority}`,
-            requestCorrelationId
-        );
+        this.logger.verbose("here-----------------------------------------------------------------------------------------------------------------");
```
Comment (sameerag) [inline lib/msal-node/src/client/ClientApplication.ts]: Can we stick to one format for logs? I do not think we had separators in general and if you want to add, would help if they are standardized.

File: lib/msal-node/src/client/ClientCredentialClient.ts [hunk 2]
```diff
@@ -229,6 +229,9 @@ export class ClientCredentialClient extends BaseClient {
                 authority.tokenEndpoint,
                 queryParametersString
             );
+
+            this.logger.info("Sending token request to host: " + authority.tokenEndpointHost);
```
Comment (pmaytak) [inline lib/msal-node/src/client/ClientCredentialClient.ts]: - Can we just print out the full tokenEndpoint, instead of just the host? Maybe helpful to have full string for debugging? Unless this is enough?
- Can this log be put near the actual call, like in `executePostToTokenEndpoint`? Unless we already log stuff there.

**Review by pmaytak — APPROVED** [review]

File: lib/msal-common/src/authority/Authority.ts [hunk 3]
```diff
@@ -237,6 +237,19 @@ export class Authority {
         }
     }
 
+    /**
+     * OAuth /token endpoint host for requests
+     */
+        public get tokenEndpointHost(): string {
```
Comment (bgavrilMS) [inline lib/msal-common/src/authority/Authority.ts]: Is it needed? Isn't `tokenEndpoint` enough?

File: lib/msal-node/src/client/ClientApplication.ts (hunk 1 above)
Comment (bgavrilMS) [reply lib/msal-node/src/client/ClientApplication.ts]: I think @trwalke meant to delete this

File: lib/msal-node/src/client/ClientApplication.ts [hunk 4]
```diff
@@ -412,10 +412,23 @@ export abstract class ClientApplication {
...
+
+        if(azureRegionConfiguration?.azureRegion)
+        {
+            this.logger.verbose(
```
Comment (bgavrilMS) [inline lib/msal-node/src/client/ClientApplication.ts]: I'd bump to info

File: lib/msal-node/src/client/ClientApplication.ts [hunk 5]
```diff
@@ -412,10 +412,23 @@ export abstract class ClientApplication {
             : this.config.auth.azureCloudOptions;
 
         // using null assertion operator as we ensure that all config values have default values in buildConfiguration()
-        this.logger.verbose(
-            `building oauth client configuration with the authority: ${authority}`,
-            requestCorrelationId
-        );
+        this.logger.verbose("here-----------------------------------------------------------------------------------------------------------------");
+
+        if(azureRegionConfiguration?.azureRegion)
```
Comment (bgavrilMS) [inline lib/msal-node/src/client/ClientApplication.ts]: Shouldn't this logging block go after line 432, where you do discovery. This way you can output "msal was configured with authority X and region Y and it uses the token endpoint Z"

**Review by bgavrilMS — APPROVED** [review]

**Review by bgavrilMS — CHANGES_REQUESTED** [review]
See comments.

File: lib/msal-node/src/client/ClientApplication.ts (hunk 1 above)
Comment (trwalke) [reply lib/msal-node/src/client/ClientApplication.ts]: Yes, not needed

File: lib/msal-common/src/authority/Authority.ts (hunk 3 above)
Comment (trwalke) [reply lib/msal-common/src/authority/Authority.ts]: `tokenEndpoint` includes the path which we consider pii from what I see in MSAL.net. I wanted a way to log it without the path. Is the path still considered pii?

Comment (trwalke) [reply lib/msal-common/src/authority/Authority.ts:243]: Is the path considered Pii?

Comment (bgavrilMS) [reply lib/msal-common/src/authority/Authority.ts]: We have discussed about PII and put some guidance here - https://learn.microsoft.com/en-us/entra/msal/dotnet/resources/handling-pii 

I don't see how the token endpoint URL can have PII. The authorization URL can have PII because it can have login_hint as query parameter, but the token endpoint cannot.

**Review by bgavrilMS — APPROVED** [review]

File: lib/msal-node/src/client/ClientApplication.ts [hunk 6]
```diff
@@ -412,17 +412,18 @@ export abstract class ClientApplication {
...
-            requestCorrelationId
-        );
         const discoveredAuthority = await this.createAuthority(
             authority,
             azureRegionConfiguration,
             requestCorrelationId,
             userAzureCloudOptions
         );
 
+        this.logger.info(
```
Comment (Robbie-Microsoft) [inline lib/msal-node/src/client/ClientApplication.ts:422]: Why did you change this from verbose to info? Additionally, why did you move this log to after the createAuthority call?

Comment (bgavrilMS) [reply lib/msal-node/src/client/ClientApplication.ts:422]: There are some many bugs / incidents where the authority that MSAL computes is wrong, it's worth having this at info level.

Comment (trwalke) [reply lib/msal-node/src/client/ClientApplication.ts:422]: Also, putting this after the discovery will allow us to get the updated authority with the regional environment if it is available.
